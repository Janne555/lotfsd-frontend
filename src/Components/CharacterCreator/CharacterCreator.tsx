import React from 'react'
import { createUseStyles } from 'react-jss'
import { useDispatch } from 'react-redux'
import { newCharacter } from '../../Redux/thunks/newCharacter'
import Attributes from '../CharacterSheet/Attributes'

const useStyles = createUseStyles((theme: Theme) => ({
  characterCreator: {
    backgroundColor: theme.colorGreyLightest,
    display: 'flex'
  },
  field: {
    
  }
}))

type Props = {

}

function CharacterCreator({ }: Props) {
  const classes = useStyles()
  const dispatch = useDispatch()

  function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
    const target = (e.target as unknown) as { elements: NewCharacterForm }
    e.preventDefault()
    dispatch(newCharacter(target.elements))
  }

  return (
    <div className={classes.characterCreator}>
      <form onSubmit={handleSubmit}>
        <div className={classes.field}>
          <label htmlFor="name">Name</label>
          <input id="name"></input>
        </div>
        <div>
          <label htmlFor="gender">Gender</label>
          <input id="gender"></input>
        </div>
        <div>
          <label htmlFor="race">Race</label>
          <input id="race"></input>
        </div>
        <div>
          <label htmlFor="alignment">Alignment</label>
          <select id="alignment">
            <option value="chaotic">Chaotic</option>
            <option value="neutral">Neutral</option>
            <option value="lawful">Lawful</option>
          </select>
        </div>
        <div>
          <label htmlFor="class">Class</label>
          <select id="class">
            <option value="fighter">Fighter</option>
            <option value="cleric">Cleric</option>
            <option value="magic-user">Magic User</option>
            <option value="specialist">Specialist</option>
          </select>
        </div>
        <Attributes attributes={{ charisma: 0, constitution: 0, dexterity: 0, intelligence: 0, strength: 0, wisdom: 0 }} />
        <button type="submit">Create</button>
      </form>
    </div>
  )
}

export default CharacterCreator