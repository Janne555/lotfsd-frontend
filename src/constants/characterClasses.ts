const characterClasses = {
  fighter: {
    savingThrows: {
      breathWeapon: 10,
      magic: 11,
      magicalDevice: 12,
      paralyze: 13,
      poison: 14
    },
    levels: [
      0, 2000, 4000, 8000, 16000, 32000, 64000
    ]
  }
}

export { characterClasses }