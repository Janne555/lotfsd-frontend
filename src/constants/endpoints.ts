const APIROOT = process.env.NODE_ENV === 'production' ? "https://lotfsdapi.azurewebsites.net/" : "http://172.17.0.2:5000/"

const ENDPOINTS = {
  LOGIN: { url: 'user/login', type: "" },
  REGISTER: { url: 'user/register', type: "" }
}

export {
  APIROOT,
  ENDPOINTS
}