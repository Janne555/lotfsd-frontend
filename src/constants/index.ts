const ATTRIBUTE_DETAILS = {
  charisma: "Retainer Recruitment, Loyalty",
  constitution: "Hit Points, Daily Travel Distance",
  dexterity: "AC, Ranged AB, Initiative",
  intelligence: "Saves vs Magic Effects",
  strength: "Mêlée AB, Open Doors",
  wisdom: "Saves vs Non-Magic Effects"
}

const SAVING_THROW_DETAILS = {
  paralyze: "Mobility Hazards (Petrification, Hold, Etc.)",
  poison: "Instant Death/KO Situations",
  breathWeapon: "Area Effects",
  magicalDevice: "Spell-Like Effects from Items",
  magic: "Spells or Innate Abilities"
}

const ATTRIBUTE_TITLES = {
  charisma: "Charisma",
  constitution: "Constitution",
  dexterity: "Dexterity",
  intelligence: "Intelligence",
  strength: "Strength",
  wisdom: "Wisdom"
}

const SAVING_THROW_TITLES = {
  paralyze: "Paralyze",
  poison: "Poison",
  breathWeapon: "Breath\xa0Weapon",
  magicalDevice: "Magical\xa0Device",
  magic: "Magic"
}

const COMMON_ACTIVITY_TITLES = {
  architecture: "Architecture",
  bushcraft: "Bushcraft",
  climbing: "Climbing",
  languages: "Languages",
  openDoors: "Open Doors",
  search: "Search",
  sleightOfHand: "Sleight of Hand",
  sneakAttack: "Sneak Attack",
  stealth: "Stealth",
  tinkering: "Tinkering"
}

const TITLES = {
  BASE_AB: "Base AB",
  MELEE_AB: "Mêlée AB",
  RANGED_AB: "Ranged AB"
}

const BASE_ARMOR_CLASS = 12

const MONEY = 'Money'

const RETAINER_KEYS: (keyof Omit<Retainer, 'uuid'>)[] = [
  "name",
  "position",
  "class",
  "level",
  "hitpoints",
  "wage",
  "share"
]

export {
  ATTRIBUTE_DETAILS,
  SAVING_THROW_DETAILS,
  ATTRIBUTE_TITLES,
  SAVING_THROW_TITLES,
  TITLES,
  BASE_ARMOR_CLASS,
  COMMON_ACTIVITY_TITLES,
  MONEY,
  RETAINER_KEYS
}