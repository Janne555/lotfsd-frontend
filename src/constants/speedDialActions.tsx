import React, { ReactNode } from 'react'
import { Icon } from '@material-ui/core'
import { match as Match } from 'react-router-dom'

type SpeedDialAction = {
  action: string | ((root: string) => string)
  icon: ReactNode | undefined
  tooltip: string
}

const speedDialActions: Record<string, SpeedDialAction[]> = {
  characters: [
    { action: root => `/characters/${root}/addItem`, icon: <Icon />, tooltip: 'Add\xa0Item' },
    { action: root => `/characters/${root}/addRetainer`, icon: <Icon />, tooltip: 'Add\xa0Retainer' },
    { action: root => `/characters/${root}/addLanguage`, icon: <Icon />, tooltip: 'Add\xa0Language' },
  ],
  itemindex: [
    { action: '/newitem', icon: <Icon />, tooltip: 'New\xa0Item' },
  ]
}

function getSpeedDialActions(match: Match<{ page: string }> | null): SpeedDialAction[] {
  const page = match?.params.page
  if (page && speedDialActions[page]) {
    return speedDialActions[page]
  } else {
    return []
  }
}


export { getSpeedDialActions }