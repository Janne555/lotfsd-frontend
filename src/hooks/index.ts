import { useDispatch, useSelector } from './redux'
import { useState, useLayoutEffect } from 'react'

function useScreenResizeEvent(condition: (screenWidth: number) => boolean) {
  const [result, setResult] = useState(condition(window.screen.width))
  useLayoutEffect(() => {
    function listener() {
      const now = condition(window.screen.width)
      if (now !== result) {
        setResult(now)
      }
    }
    window.addEventListener('resize', listener)
    return () => window.removeEventListener('resize', listener)
  })
  return result
}

export {
  useDispatch,
  useSelector,
  useScreenResizeEvent
}