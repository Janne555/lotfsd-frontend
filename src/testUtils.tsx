import fs from 'fs'

async function writeComponent(htmlElement: HTMLElement) {
  const html = `
    <html>
      <body>
        ${htmlElement.innerHTML}
      </body>
    </html>
  `
  fs.writeFile('debugging/output.html', html, console.log)
}

export {
  writeComponent
}