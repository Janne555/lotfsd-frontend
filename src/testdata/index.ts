import { ItemsQuery } from "../../__generated__/apolloTypes/ItemsQuery"
import { MockedResponse } from "@apollo/react-testing"
import { CharacterListQuery } from "../../__generated__/apolloTypes/CharacterListQuery"
import { CHARACTER_LIST_QUERY, ITEMS_QUERY } from "../constants"

const ATTRIBUTES: Attributes = {
  charisma: 10,
  constitution: 11,
  dexterity: 12,
  intelligence: 13,
  strength: 14,
  wisdom: 15
}

const WALLET: Wallet = {
  copper: 10,
  gold: 20,
  silver: 30
}

const EQUIPMENT: EquipmentListItem[] = [
  {
    encumbrancePoints: 1,
    amount: 4,
    name: "thingy",
    itemId: "1",
    listItemId: "2",
    stackSize: 4
  },
  {
    encumbrancePoints: 4,
    amount: 4,
    name: "fooby",
    itemId: "3",
    listItemId: "4",
    stackSize: 4
  },
  {
    encumbrancePoints: 1,
    amount: 4,
    name: "baar",
    itemId: "5",
    listItemId: "6",
    stackSize: 4,
    encumbrance: 1
  }
]

type Result<T> = {
  data: T
}

const CHARACTER_LIST_MOCK: Result<CharacterListQuery> = {
  data: {
    characterSheets: [
      {
        __typename: "CharacterSheet",
        class: "fighter",
        id: "213123",
        name: "zoid"
      }
    ]
  }
}

const ITEM_INDEX_MOCK: Result<ItemsQuery> = {
  data: {
    items: [
      {
        __typename: "Item",
        attackBonus: null,
        baseArmorClass: null,
        damage: null,
        description: "a description",
        effects: null,
        encumbrance: null,
        encumbrancePoints: 1,
        id: "asd",
        name: "thingy",
        rangeLong: null,
        rangeMedium: null,
        rangeShort: null,
        stackSize: 1,
        type: "item"
      }
    ]
  }
}

const APOLLO_MOCKS: MockedResponse[] = [
  {
    request: {
      query: CHARACTER_LIST_QUERY
    },
    result: CHARACTER_LIST_MOCK
  },
  {
    request: {
      query: ITEMS_QUERY
    },
    result: ITEM_INDEX_MOCK
  }
]

export {
  ATTRIBUTES,
  WALLET,
  EQUIPMENT,
  APOLLO_MOCKS
}