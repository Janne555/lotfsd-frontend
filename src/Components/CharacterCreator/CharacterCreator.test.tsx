import React from 'react'
import { JssProvider } from 'react-jss'
import { render, fireEvent } from '@testing-library/react'
import { MockedProvider, wait, MockedResponse } from '@apollo/react-testing'
import { MemoryRouter } from 'react-router-dom'
import { LoginProvider } from '../../hooks'
import CharacterCreator from './CharacterCreator'
import userEvent from '@testing-library/user-event'
import { writeComponent } from '../../testUtils'
import { APOLLO_MOCKS } from '../../testdata'
import { CREATE_CHARACTER_MUTATION, CHARACTER_LIST_QUERY } from '../../constants'
import { InMemoryCache } from 'apollo-boost'
import { act } from 'react-dom/test-utils'
import { CharacterListQuery } from '../../../__generated__/apolloTypes/CharacterListQuery'

describe('<CharacterCreator />', () => {
  it('should match snapshot', async () => {
    const { asFragment } = render(
      <MemoryRouter>
        <LoginProvider>
          <MockedProvider mocks={[]}>
            <JssProvider disableStylesGeneration>
              <CharacterCreator />
            </JssProvider>
          </MockedProvider>
        </LoginProvider>
      </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot()
  });

  it('should create a character', async () => {
    const newData = jest.fn(() => ({
      data: {
        createCharacterSheet: {
          __typename: "CharacterSheet",
          class: "Cleric",
          name: "Zoid",
          id: "asd"
        }
      }
    }))

    const mocks: MockedResponse[] = [
      {
        request: {
          query: CREATE_CHARACTER_MUTATION,
          variables: { "characterSheet": { "charisma": 10, "dexterity": 12, "intelligence": 13, "strength": 14, "wisdom": 15, "constitution": 11, "copper": 0, "gold": 0, "silver": 100, "alignment": "chaotic", "class": "fighter", "gender": "Male", "name": "Zoid", "race": "Crab", "age": 20, "maxHp": 10, "attackBonus": 1, "currentHp": 10, "experience": 0, "surpriseChance": 4, "defensive": true, "improvedParry": true, "press": true, "standard": true, "parry": false, "architecture": 1, "bushcraft": 1, "climbing": 1, "languages": 1, "openDoors": 1, "search": 1, "sleightOfHand": 1, "sneakAttack": 1, "stealth": 1, "tinkering": 1, "breathWeapon": 10, "magic": 11, "magicalDevice": 12, "paralyze": 13, "poison": 14 } }
        },
        result: {
          data: {
            createCharacterSheet: {
              __typename: "CharacterSheet",
              class: "Cleric",
              name: "Zoid",
              id: "asd"
            }
          }
        },
        newData
      }
    ]

    const cache = new InMemoryCache()

    cache.writeQuery<CharacterListQuery, string>({
      query: CHARACTER_LIST_QUERY,
      data: {
        characterSheets: []
      }
    })

    const { getByLabelText, getByText } = render(
      <MemoryRouter>
        <LoginProvider>
          <MockedProvider mocks={mocks} cache={cache} >
            <JssProvider disableStylesGeneration>
              <CharacterCreator />
            </JssProvider>
          </MockedProvider>
        </LoginProvider>
      </MemoryRouter>
    )

    userEvent.type(getByLabelText("Charisma"), "10")
    userEvent.type(getByLabelText("Constitution"), "11")
    userEvent.type(getByLabelText("Dexterity"), "12")
    userEvent.type(getByLabelText("Intelligence"), "13")
    userEvent.type(getByLabelText("Strength"), "14")
    userEvent.type(getByLabelText("Wisdom"), "15")
    userEvent.type(getByLabelText(/Name/), "Zoid")
    userEvent.type(getByLabelText(/Gender/), "Male")
    userEvent.type(getByLabelText(/Race/), "Crab")
    userEvent.type(getByLabelText(/Age/), "20")
    userEvent.type(getByLabelText(/Money/), "100")
    userEvent.type(getByLabelText(/HP/), "10")
    fireEvent.click(getByText("Create"))

    await act(async () => {
      await wait(100)
    })

    expect(newData).toHaveBeenCalled()
  });
});