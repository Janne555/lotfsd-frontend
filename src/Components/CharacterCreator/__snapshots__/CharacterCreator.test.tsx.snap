// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterCreator /> should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <div>
        <span>
          Ability Scores
        </span>
        <span>
          Modifiers
        </span>
        <label
          for="charisma"
        >
          Charisma
        </label>
        <div>
          <input
            id="charisma"
            type="number"
            value="0"
          />
        </div>
        <div>
          <span>
            -3
          </span>
        </div>
        <span>
          Retainer Recruitment, Loyalty
        </span>
        <label
          for="constitution"
        >
          Constitution
        </label>
        <div>
          <input
            id="constitution"
            type="number"
            value="0"
          />
        </div>
        <div>
          <span>
            -3
          </span>
        </div>
        <span>
          Hit Points, Daily Travel Distance
        </span>
        <label
          for="dexterity"
        >
          Dexterity
        </label>
        <div>
          <input
            id="dexterity"
            type="number"
            value="0"
          />
        </div>
        <div>
          <span>
            -3
          </span>
        </div>
        <span>
          AC, Ranged AB, Initiative
        </span>
        <label
          for="intelligence"
        >
          Intelligence
        </label>
        <div>
          <input
            id="intelligence"
            type="number"
            value="0"
          />
        </div>
        <div>
          <span>
            -3
          </span>
        </div>
        <span>
          Saves vs Magic Effects
        </span>
        <label
          for="strength"
        >
          Strength
        </label>
        <div>
          <input
            id="strength"
            type="number"
            value="0"
          />
        </div>
        <div>
          <span>
            -3
          </span>
        </div>
        <span>
          Mêlée AB, Open Doors
        </span>
        <label
          for="wisdom"
        >
          Wisdom
        </label>
        <div>
          <input
            id="wisdom"
            type="number"
            value="0"
          />
        </div>
        <div>
          <span>
            -3
          </span>
        </div>
        <span>
          Saves vs Non-Magic Effects
        </span>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Randomize
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
          for="name"
          id="name-label"
        >
          Name
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="name"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
          for="gender"
          id="gender-label"
        >
          Gender
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="gender"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
          for="race"
          id="race-label"
        >
          Race
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="race"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
          for="age"
          id="age-label"
        >
          Age
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="age"
            required=""
            type="number"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
          for="money"
          id="money-label"
        >
          Money
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="money"
            required=""
            type="number"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
          data-shrink="false"
          for="maxHp"
          id="maxHp-label"
        >
          HP
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input"
            id="maxHp"
            required=""
            type="number"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
          data-shrink="true"
          for="alignment"
        >
          Alignment
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <select
            aria-invalid="false"
            class="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input"
            id="alignment"
            required=""
          >
            <option
              value="chaotic"
            >
              Chaotic
            </option>
            <option
              value="neutral"
            >
              Neutral
            </option>
            <option
              value="lawful"
            >
              Lawful
            </option>
          </select>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiNativeSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
          data-shrink="true"
          for="class"
        >
          Class
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <select
            aria-invalid="false"
            class="MuiNativeSelect-root MuiNativeSelect-select MuiInputBase-input MuiInput-input"
            id="class"
            required=""
          >
            <option
              value="fighter"
            >
              Fighter
            </option>
            <option
              value="cleric"
            >
              Cleric
            </option>
            <option
              value="magic-user"
            >
              Magic User
            </option>
            <option
              value="specialist"
            >
              Specialist
            </option>
          </select>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiNativeSelect-icon"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Create
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <p />
    </form>
  </div>
</DocumentFragment>
`;
